### PR TITLE
IDEMPIERE-5944 - Default value of Yes/No columns

### PR DESCRIPTION
**Pull request by VitorVilla**
https://idempiere.atlassian.net/browse/IDEMPIERE-5944

# Pull Request Checklist

- [ ] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [ ] My code follows the best practices of this project
- [ ] I have performed a self-review of my own code
- [ ] My code is easy to understand and review. 
- [ ] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [ ] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Comment by VitorVilla** [comment]
As discussed in the mattermost, it was implemented in MColumn.beforeSave to make it always mandatory and alert the Dev to set a default value. Additionally, an Update was performed to standardize the relevant fields.

File: org.adempiere.base/src/org/compiere/model/MColumn.java [hunk 1]
```diff
@@ -561,6 +561,11 @@ && getAD_Element_ID() != 0)
 				return false;
 			}
 		}
+		
+		if (get_ValueAsInt(COLUMNNAME_AD_Reference_ID) == 20 && (!get_ValueAsBoolean(COLUMNNAME_IsMandatory) || get_Value(COLUMNNAME_DefaultValue) == null)) {
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MColumn.java:565]: should be `if (getAD_Reference_ID()==DisplayType.YesNo) && (!isMandatory() || Util.isEmpty(getDefaultValue(),true))` instead.

**Comment by hengsin** [comment]
hi @CarlosRuiz-globalqss , is this what's agree upon in the discussion ?

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Hi @VitorVilla 

Can you please test the following change:
[PR2121_pr.patch.txt](https://github.com/idempiere/idempiere/files/13589876/PR2121_pr.patch.txt)

Is a minor rewrite of what you did, but also ensures that mandatory is set properly, even if the default is set.

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**